src/function_file.py

```diff
@@ -62,13 +54,16 @@ def zip_and_upload_folder(client: CogniteClient, fn_config: FunctionConfig, xid:
     logger.info(f"Uploading code from '{fn_config.function_folder}' to Files using external ID: '{xid}'")
     buf = io.BytesIO()  # TempDir, who needs that?! :rocket:
     with ZipFile(buf, mode="a") as zf:
-        with temporary_chdir(fn_config.function_folder):
-            _write_files_to_zip_buffer(zf, directory=Path())
-
-        if (common_folder := fn_config.common_folder) is not None:
-            with temporary_chdir(common_folder.parent):  # Note .parent
-                logger.info(f"- Added common directory: '{common_folder}' to the file/function")
-                _write_files_to_zip_buffer(zf, directory=common_folder)
+        function_folder = Path(fn_config.function_folder)
+        for filepath in function_folder.rglob("*"):
+            dest_filepath = str(filepath).replace(str(function_folder), "")
+            zf.write(filepath, dest_filepath)
+
+        if fn_config.common_folder:
+            common_folder = Path(fn_config.common_folder)
+            for filepath in common_folder.rglob("*"):
+                dest_filepath = f'common/{str(filepath).replace(str(common_folder), "")}'
```

Comment:
People may pass the name of the common folder with `common_folder`, won't this ignore the setting? And with your changes, assuming someone is passing, say, `common_folder=foo/bar/baz`, wouldn't `common/` also be wrong? 🤔 
```py
dest_filepath = f'common/{str(filepath).replace(str(common_folder), "")}'
                  ^^^^^^
```

Reply:
The difficult part is how does the user structure their scripts to import a module from the common_folder for both local testing and as a deployed function. Do you think it would be a good idea to make it a hard requirement that the common_folder must be in the same directory as `function_folder`? Functions that use a "common" module like this should already have their imports funky:
```
import sys
sys.path.append('..')
from ../common import mymod
```
Then I could force this:
1. common_folder should be full path
2. common_folder will be at same level as function_folder in zip

Reply:
I do feel that this change comes at a great cost to complexity; is your use case (nested common folder) _really_ worth it? (I feel I have to ask 😬 )

It might be better to make a different function-action repo for this purpose maybe?

Reply:
Yes it is _really_ worth it. I thought it was non-nested because someone couldn't figure it out, not because it was a design choice? The idea is to get this action to be flexible and allow more than just functions in a repo.
Your comment to remove the parent common directory from the zip is a valid one. I shouldn't change its location in the zip. I'll update this PR soon.

Reply:
Seems you guys are determined that _nested common-folder structures_ is valuable, so I won't block this 😄 

Will review once @petreeb addresses the currently hardcoded `f'common/` comment above 👍

Reply:
@Magssch it was described here:
----
<img width="849" alt="Screenshot 2024-01-29 at 10 02 51" src="https://github.com/cognitedata/function-action-oidc/assets/8521241/a49a8364-5b07-4ea9-874e-1f44d35b8680">

Reply:
One last comment though, we need to make sure that running locally and in Functions works as expected - especially with regards to `import`s

Reply:
Ok I've fixed this part. I've tested with common both inside and outside the function. I am going to update the README tomorrow